File: scala/parsers/src/test/scala/jp/co/teruuu/parser/common/StringParserTest.scala
package jp.co.teruuu.parser.common

import org.scalatest.FlatSpec;

class StringParserTest extends FlatSpec {

  "parse" should "successt" in {
    val parse = Parser.string("Hello")
    val result1 = parse.parse("Hello, world")

    assert(result1.isInstanceOf[ParseOk[_]])
    assert(result1.asInstanceOf[ParseOk[_]].next.equals(", world"))
  }

  "parse" should "failer" in {
    val parse = Parser.string("Hello")
    val result1 = parse.parse("hello, world")

    assert(result1.isInstanceOf[ParseNg[_]])
  }
}

File: java/parsers_java8/src/test/java/jp/co/teruuu/parser/common/StringParserTest.java
package jp.co.teruuu.parser.common;


import org.junit.Test;

public class StringParserTest {

    @Test
    public void test() {
        Parser<String> strParser1 = Parser.string("Hello");
        ParseResult<String> result1 = strParser1.parse("HelloWorld");
        assert(result1 instanceof ParseResult.Success);

        ParseResult<String> result2 = strParser1.parse("Hello, World");
        assert(result2 instanceof ParseResult.Success);

        ParseResult<String> result3 = strParser1.parse("hello, World");
        assert(result3 instanceof ParseResult.Failure);

    }
}
